### PR TITLE
feat: annotation table updates

### PR DESCRIPTION
**Pull request by codemonkey800**
#432

- Adds rudimentary feature flagging module that can be controlled via code or query parameter
- Updates the annotation table to match the updated design
  - Implements but hides method type column

## Demo

https://dev-jeremy-anno-table.cryoet.dev.si.czi.technology/runs/242

image

### With Method Type

image

### CODE REVIEW

File: frontend/packages/data-portal/app/components/Run/AnnotationObjectTable.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ export function AnnotationObjectTable() {
         },
         {
           label: t('objectShapeType'),
-          values: [annotation.shape_type],
+          values: [annotation.files[0].shape_type],
```
Comment (kne42) [inline frontend/packages/data-portal/app/components/Run/AnnotationObjectTable.tsx:38]: why this change? did the data model itself change?

File: frontend/packages/data-portal/app/utils/featureFlags.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export const FEATURE_FLAGS = {
+  methodType: 'dev',
+} as const
+
+export type FeatureFlagKey = keyof typeof FEATURE_FLAGS
+
+const ENABLE_FEATURE_PARAM = 'enable-feature'
+const DISABLE_FEATURE_PARAM = 'disable-feature'
+
+export function getFeatureFlag({
```
Comment (kne42) [inline frontend/packages/data-portal/app/utils/featureFlags.ts:16]: this seems more like a user setting than anything – would we be accomplishing this stuff with cookies in the future?

**Review by kne42 — APPROVED** [review]
LGTM, had some questions

**Comment by Janeece** [comment]
@codemonkey800 sorry I'm late to the party on this one, i have one comment — the table cell link for the method type should not be the minimal button, should just be a normal text link. 
Screenshot 2024-03-07 at 4 57 19 PM

Not a critical-blocker but it's over emphasized + confusing in how it competes with the action buttons on the right. Would love to see this updated if you have the time, otherwise logged as a bug to fix later.

File: frontend/packages/data-portal/app/utils/featureFlags.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { useSearchParams } from '@remix-run/react'
+
+import { useEnvironment } from 'app/context/Environment.context'
+
+export type FeatureFlagEnvironment = typeof process.env.ENV
+
+export const FEATURE_FLAGS = {
+  methodType: 'dev',
```
Comment (manasaV3) [inline frontend/packages/data-portal/app/utils/featureFlags.ts:8]: We should update the data structure so that we can enable a feature flag for multiple environments.
